### ISSUE TITLE
Running `neovide --version` fails on Linux if DISPLAY or WAYLAND_DISPLAY is not set

### ISSUE THREAD

**Issue by tadams42** [Kethku/neovide#2867]
**Describe the bug**

Running `neovide --version` in context where neither `$DISPLAY` nor `$WAYLAND_DISPLAY` are defined panicks.

**To Reproduce**

1. `sudo su -` 
   -  this way neither `$DISPLAY` nor `$WALYAND_DISPLAY` are no longer defined
2. `neovide --version`
3. See error

Alternatively (and this is how I originally encountered this same error) we can try querying `neovide` version from within Python script like this:

```py
import subprocess

data = subprocess.check_output(
    ["/usr/local/bin/neovide", "--version"], 
    shell=False, 
    encoding="utf-8"
)
```

Both approaches give exact same error: Python is not strictly needed to reproduce this.

**Expected behavior**

Output like this:

```
neovide 0.13.3
```

It is OK to depend on X11/Wayland display variable and to refuse to run without them. But it would also be preferable that simple, commandline version query works even without Wayland/X11.

**Screenshots**

N/A

**Desktop (please complete the following information):**
 - OS: `Ubuntu 24.04.1 LTS` `Linux 6.8.0-48-generic #48-Ubuntu SMP PREEMPT_DYNAMIC Fri Sep 27 14:04:52 UTC 2024 x86_64 x86_64 x86_64` `GNU/Linux`
 - Neovide Version: `0.13.3`
 - Neovim Version: `NVIM v0.10.3-dev-13+gb286ba419a Build type: RelWithDebInfo LuaJIT 2.1.1713484068`

**Please run `neovide --log` and paste the contents of the `.log` file created in the current directory here:**

```sh
# neovide --version
Neovide panicked with the message 'Failed to create winit event loop: Os(OsError { line: 786, file: "/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/winit-0.29.15/src/platform_impl/linux/mod.rs", error: Misc("neither WAYLAND_DISPLAY nor WAYLAND_SOCKET nor DISPLAY is set.") })'. (File: src/window/mod.rs; Line: 126, Column: 38)
This is a bug and we would love for it to be reported to https://github.com/neovide/neovide/issues

Backtrace saved to neovide_backtraces.log!
```

```sh
# neovide --log
Neovide panicked with the message 'Failed to create winit event loop: Os(OsError { line: 786, file: "/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/winit-0.29.15/src/platform_impl/linux/mod.rs", error: Misc("neither WAYLAND_DISPLAY nor WAYLAND_SOCKET nor DISPLAY is set.") })'. (File: src/window/mod.rs; Line: 126, Column: 38)
This is a bug and we would love for it to be reported to https://github.com/neovide/neovide/issues

Backtrace saved to neovide_backtraces.log!
```

```sh
# cat neovide_backtraces.log
2024-11-08 16:11:57 - Neovide panicked with the message 'Failed to create winit event loop: Os(OsError { line: 786, file: "/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/winit-0.29.15/src/platform_impl/linux/mod.rs", error: Misc("neither WAYLAND_DISPLAY nor WAYLAND_SOCKET nor DISPLAY is set.") })'. (File: src/window/mod.rs; Line: 126, Column: 38)
   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>
   7: <unknown>
   8: <unknown>
   9: <unknown>
  10: __libc_start_call_main
             at ./csu/../sysdeps/nptl/libc_start_call_main.h:58:16
  11: __libc_start_main_impl
             at ./csu/../csu/libc-start.c:360:3
  12: <unknown>
```

**Additional context**
N/A

**Comment by fredizzimo** [comment]
It's strongly discouraged to run Neovide from root account, but if you really need to then you can check these instructions out
https://wiki.archlinux.org/title/Running_GUI_applications_as_root#Wayland

A much better approach is to use `sudoedit` https://www.man7.org/linux/man-pages/man8/sudoedit.8.html

`EDITOR=neovide` sudoedit path/to/file`

**Comment by tadams42** [comment]
Well, that's the thing: I don't want to edit file as sudo here, I just want to query `neovide` version. 

`sudo` example above was only to create runtime context in which there is not X11/Wayland available.

**Comment by fredizzimo** [comment]
In which context do you need to run `neovide --version`, where you don't have a GUI available?

The fix for this would be quite tricky, since we also need to be able to show command-line output in a GUI window, when a terminal is attached.  Which means that we need differents code paths for creating and managing the event loop. It's not impossible to do, but we are in the middle of some big refactoring to support multiple windows on macOS, so this has to wait until that is done.

BTW, you could run `env -u WAYLAND_DISPLAY env -u DISPLAY neovide --version` instead of sudo to demonstrate the problem.

**Comment by tadams42** [comment]
> The fix for this would be quite tricky...

Understood! 

Also, and more importantly: I'd just tried getting a version for few other GUI (mostly KDE) apps in Linux TTY. They all have problems when no display is available. Therefore, I think I'm requesting an exotic "thing" no one else has either.

With all that said, please close this issue with something like "won't fix".

And thank you very much for looking into this! 

> BTW, you could run env -u WAYLAND_DISPLAY env -u DISPLAY neovide --version instead of sudo to demonstrate the problem.

You are absolutely right! And, I could've also gone to Linux TTY and do it there, without throwing `sudo` in the mix.
Sorry for all the confusion I'd caused with that `sudo` example :-( ...

> In which context do you need to run neovide --version, where you don't have a GUI available?

I'd written bunch of Python scripts that pull some apps directly from GitHub releases into my `/usr/local/bin`. 
Before downloading stuff from GitHub, my code queries local app for it's version and compares that to latest GitHub release. 
It lets me have all the newest goodies without messing the rest of my system but also without relying on (slow) updates through official distro channels.

One of the apps I'm pulling is `neovide`. And I'm trying to query it's version via `Python subprocess` module. 
Without shell and without display. And yes, `sudo` is also involved, but that is beside the point. 

Anyway, thanks again for looking into this!
